examples/apps/ping/Makefile

```diff
@@ -1,19 +1,36 @@
+specs = $(wildcard specs/*.rflx)
+src = generated/rflx.ads
+bin = obj/ping
+generate = rflx generate -d generated $(specs)
+build = gprbuild -Pping
+
+.PHONY: test test_python test_spark build prove generate clean clean_proof
+
```

Comment:
This defeats the purpose of `make` and essentially uses it as a poor shell script. I think it would be better to make the sources a result of the generation step, make the build step depend on it and get rid of the phony targets:

```make
generated/rflx.ads: $(specs)
   rflx generate -d generated $^

$(bin): generated/rflx.ads
   gprbuild -Pping
```

Reply:
Could you please explain how your suggested change would improve the behavior? The Makefile was a "poor shell script" before and certainly still is to a certain extent after my change, but in the current state it works I would expect it: The code is only generated if the generated code (represented by `rflx.ads`) is missing or any specification file in `spec` is changed. Removing the `generate` target would remove an easy way to regenerate the code after a change to the code generate, which I used often lately. I don't see that it makes sense to get rid of any phony target.

Reply:
Changed in c7b639e as discussed separately.